File: Lab/Parser-3/Assign.cpp
#include "Assign.h"

extern Tokenizer t;

int resultAssign=0;
void Assign::parseAssign()
{
  curToken=t.getToken();
  //id=ID->parseIDForSS();

  if(curToken != 14) { cout<<"Error-expecting = "<<endl; exit(1); }
  t.skipToken();
  exp.parseExpression(); 

  if(curToken != 12) { cout<<"Error-expecting ; "<<endl; exit(1); }
  t.skipToken();
  //id=ID->parseIDForSS();
}

void Assign::printAssign()
{
  //cout<<id.getName();
  exp.printExpression();
  cout<<" ; ";
}

void Assign::execAssign()
{
  resultAssign=exp.execExpression();
  id.setIDVal(resultAssign);
}

File: Lab/Lab2/Aniket/interpreter/Assign.java
import java.io.*;

//class for <assign> NT
class Assign {
    Id id;
    Exp exp;
    public Assign() {
        id = null;
        exp = null;
    }
    public void parseAssign() throws FileNotFoundException, IOException {
        id = Id.parseIdForSS();

        //check & skip terminal "="
        if(14 != Tokenizer.instance().getToken()) {
            System.out.println("Parse error : expected : =");
            System.exit(0);
        }
        Tokenizer.instance().skipToken();
        
        exp = new Exp();
        exp.parseExp();

        //check & skip terminal ;
        if(12 != Tokenizer.instance().getToken()) {
            System.out.println("Parse error : expected ;");
            System.exit(0);
        }
        Tokenizer.instance().skipToken();
    }

    public void printAssign(int indentLevel) throws IOException {
        System.out.print(String.format("%1$#" + (4*indentLevel) + "s"+id.getName()+" = ", " "));
        exp.printExp();
        System.out.println(";");
    }

    public void executeAssign() throws FileNotFoundException,IOException {
        id.setValue(exp.evalExp());
    }
}
